OpenEdXMobile/src/main/java/org/edx/mobile/module/db/impl/DbHelper.java

```diff
@@ -77,98 +77,102 @@ private void createAssessmentTable(SQLiteDatabase db){
 
     @Override
     public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        db.beginTransaction();
```

Comment:
This scoping is not correct. You're starting the transaction from the start, but only marking it as successful (unconditionally) in the specific upgrade to version 6. This would mean that the transaction would fail on subsequent future upgrades.

What you need to do is to only begin (and end) the nested transaction in the version 6 upgrade specifically, and only mark it as successful if the migration is truly successful (i.e. the external storage is mounted, and the directory move and database upgrades are successful.